### ISSUE TITLE
Unite element-hints and element-frame

### ISSUE THREAD

**Issue by aartaka** [atlas-engineer/nyxt#1651]
This started out as [a Reddit idea](https://www.reddit.com/r/Nyxt/comments/ofq4qt/select_frame_a_new_model_for_interaction/h4g46rv?utm_source=share&utm_medium=web2x&context=3) and it haunts me ever since: how about we unite element hints and element frame into one selection method? The benefits are:
- Users can choose whatever input device/model they like, while still being hella productive.
- DOM parsing we do for element hints can make element frame much more customizable and tailored to the task.
- We will have more of a cohesion in commands -- alfter all, doing two methods (hints & frame) requires writing twice as more commands (think `select-frame-new-buffer` and `follow-hint-new-buffer` as examples of the same behavior with only the input differing) compared to one unified method.

How does it sound?

**Comment by jmercouris** [comment]
Element frame is a source, but element hint is more than just a source. I don't know how I would combine them into a unified intreface. There may be something here, but I think we should think a lot more about it before integration.

**Comment by aadcg** [comment]
As of today `hint` is now a mode and `element-frame` was absorbed into `document` (previously `web`).

I actually agree that element frame is a source and it should be integrated into `hint`. 

But perhaps @Ambrevar remembers why we went with 7fa97ae6f06a0b30103dc3cd102d9e87a07caa74.

**Comment by Ambrevar** [comment]
Yes, because it used to be in web-mode, which is now document mode.  It was the simplest route.  https://github.com/atlas-engineer/nyxt/commit/7fa97ae6f06a0b30103dc3cd102d9e87a07caa74 is by no mean a final decision.

If we want to make it part of hint-mode, then expedition mode should be a submode of hint-mode I suppose.

Pull requests welcome!

**Comment by aartaka** [comment]
> If we want to make it part of hint-mode, then expedition mode should be a submode of hint-mode I suppose.

Quite like `visual-mode` :D

**Comment by Ambrevar** [comment]
Indeed!

**Comment by aadcg** [comment]
Hmm, then `hint` doesn't seem very good at expressing that it's the parent of `visual-mode`, the link hints and `element-frame`. Perhaps `element`?

**Comment by aartaka** [comment]
In the ideal world, maybe. But in our case, it's all link hints-centered :)

**Comment by aartaka** [comment]
It's centered on hints -- you basically choose the area of the page using the hints infrastructure!

**Comment by aadcg** [comment]
It seems I misread your previous message. Perhaps `link` was extra there?

**Comment by aartaka** [comment]
Yes, "link" was unnecessary.

**Comment by aadcg** [comment]
I can't work on this for 3.0.

**Comment by aadcg** [comment]
Is this a priority for 3.0?

**Comment by jmercouris** [comment]
It is not.

**Comment by aartaka** [comment]
I'd suggest yet another radical move: remove the public API for element frames from Nyxt. Not necessarily deleting all the code, but at least commenting it out (with the note to re-enable and merge with hint-mode). Or at least commenting the broken commands out. Because if we have commands that don't work, it's a bug.

**Comment by jmercouris** [comment]
I am strongly against that, it should be fixed for 3.0.0.

**Comment by jmercouris** [comment]
That is correct. It should be functional, but it need not be united for 3.0.0.

**Comment by aartaka** [comment]
> @aartaka could you please open a separate issue detailing which functionality is broken?

It's been there for, like, a year now: #2293

**Comment by jmercouris** [comment]
I'd rather avoid unification, until the code is significantly debugged and robust. I would like to see a very reliable solution, before we take the models and integrate them.
